src/screens/PinCodeUnlock/PinCodeUnlock.js

```diff
@@ -115,7 +122,21 @@ class PinCodeUnlock extends React.Component<Props, State> {
         this.loginWithPrivateKey(data);
       }).catch(this.requirePinLogin);
     }
-  }
+  };
+
+  handleUnlockAction = ({ pin, privateKey, defaultAction }: HandleUnlockActionProps) => {
+    const { navigation, proceedWithPk } = this.props;
+    const customUnlockAction = navigation.getParam('customUnlockAction');
+    if (customUnlockAction && typeof customUnlockAction === 'function') {
+      if (privateKey) {
+        customUnlockAction(privateKey);
+      } else if (pin) {
+        proceedWithPk(pin, customUnlockAction);
+      }
+    } else {
+      defaultAction();
+    }
+  };
```

Comment:
Can we move this logic to `authActions`? This screen has been crazy enough...
Also see my prev comment about `customUnlockAction`. Ideally, we should just pass a boolean value to `handleUnlockAction` and handle all the logic outside of this screen.

Reply:
Moved :)